.github/styles/Google/EnDash.yml

```diff
@@ -0,0 +1,13 @@
+extends: existence
+message: "Use an em dash ('—') instead of '–'."
```

Comment:
these 2 dashes look the same to me...how does one even do an em dash?

Reply:
They look the same in monospace, but they look different in other typefaces.

<img width="47" alt="image" src="https://github.com/ray-project/ray/assets/26107013/7b56c1ee-8e9a-4ecb-b67d-409206499902">

@angelinalg what're your thoughts on this rule?

Reply:
> how does one even do an em dash

Also @amogkam to answer your question, I just google "em dash" and then copy it lol

<img width="1096" alt="image" src="https://github.com/ray-project/ray/assets/26107013/ed966641-a31c-44a1-a65b-f34e61d03965">

Reply:
Okay, did some more research. Sphinx renders two normal dashes (`--`) as a en dash (–) and three normal dashes (`---`) as an em dash (—). 

I've updated the Vale rules accordingly.